### PR TITLE
Added "Remove file?" confirmation modal to CST file upload forms

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- Added remove file confirmation modal to these files:
  - `src/applications/claims-status/components/AddFilesFormOld.jsx`
  - `src/applications/claims-status/components/claim-files-tab/AddFilesForm.jsx`

## Related issue(s)

- [[Comprehension] User is uncertain where a button or link leads.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80604)
- Design mock: https://www.figma.com/file/F8U4wddaFouUPVd4mGBMDI/CST---Evidence-Submission---Spec?type=design&node-id=1712-10002&mode=design&t=EPlgtXxIz607wYDP-0

## Testing done

- Manually tested each modal

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | Screenshot 2024-04-17 at 12 52 10 PM | Screenshot 2024-04-18 at 12 46 54 PM |
| Desktop | Screenshot 2024-04-17 at 12 51 05 PM | Screenshot 2024-04-18 at 12 46 31 PM |

Screenshot 2024-04-18 at 1 03 25 PM

## What areas of the site does it impact?

CST Add Evidence and Document Request forms.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by jerekshoe — DISMISSED** [review]

File: src/applications/claims-status/components/AddFilesFormOld.jsx [hunk 1]
```diff
@@ -162,6 +165,14 @@ class AddFilesFormOld extends React.Component {
     this.props.onDirtyFields();
   };
 
+  removeFileConfirmation = (fileIndex, fileName) => {
```
Comment (pmclaren19) [inline src/applications/claims-status/components/AddFilesFormOld.jsx:168]: @samcoforma please add a unit test for this functionality!

File: src/applications/claims-status/components/AddFilesFormOld.jsx [hunk 2]
```diff
@@ -218,7 +229,9 @@ class AddFilesFormOld extends React.Component {
                     <va-button
                       secondary
                       text="Remove"
-                      onClick={() => this.props.onRemoveFile(index)}
+                      onClick={() => {
```
Comment (pmclaren19) [inline src/applications/claims-status/components/AddFilesFormOld.jsx:232]: Please add a unit test around when a user clicks the button what happens

File: src/applications/claims-status/components/claim-files-tab/AddFilesForm.jsx [hunk 3]
```diff
@@ -278,6 +291,40 @@ class AddFilesForm extends React.Component {
         >
           {mailMessage}
         </va-additional-info>
+        <VaModal
```
Comment (pmclaren19) [inline src/applications/claims-status/components/claim-files-tab/AddFilesForm.jsx:294]: @samcoforma please add a unit test for this new modal functionality!

File: src/applications/claims-status/components/AddFilesFormOld.jsx [hunk 4]
```diff
@@ -291,6 +304,40 @@ class AddFilesFormOld extends React.Component {
...
+              showRemoveFileModal: false,
+              removeFileIndex: null,
+              removeFileName: null,
+            });
+          }}
+          primaryButtonText="Remove"
+          secondaryButtonText="Keep"
+          visible={this.state.showRemoveFileModal}
+        >
+          <h3>Remove this file?</h3>
```
Comment (pmclaren19) [inline src/applications/claims-status/components/AddFilesFormOld.jsx:335]: update this to be the prop modalTitle

File: src/applications/claims-status/components/claim-files-tab/AddFilesForm.jsx [hunk 5]
```diff
@@ -278,6 +291,40 @@ class AddFilesForm extends React.Component {
...
+              showRemoveFileModal: false,
+              removeFileIndex: null,
+              removeFileName: null,
+            });
+          }}
+          primaryButtonText="Remove"
+          secondaryButtonText="Keep"
+          visible={this.state.showRemoveFileModal}
+        >
+          <h3>Remove this file?</h3>
```
Comment (pmclaren19) [inline src/applications/claims-status/components/claim-files-tab/AddFilesForm.jsx:322]: update this to be the prop modalTitle

**Review by powellkerry — DISMISSED** [review]

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    const { getByText } = render(
+      <RemoveFileModal
+        removeFileName={removeFileName}
+        showRemoveFileModal={showRemoveFileModal}
+        closeModal={onCloseModal}
+        removeFile={onRemoveFile}
+      />,
+    );
+
+    expect(getByText(removeFileName)).to.exist;
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:49]: I would combine this with the render test above, no need to have two different tests here!

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+import { render } from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { $ } from '@department-of-veterans-affairs/platform-forms-system/ui';
+
+import RemoveFileModal from '../../../components/claim-files-tab/RemoveFileModal';
+
+describe('<RemoveFileModal>', () => {
+  it('should render component', () => {
+    const removeFileName = 'test.pdf';
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:11]: id make a props const that has all of the props for remove file modal that way you dont have to set them in each test ex:

const props = {
removeFileName: 'test.pdf',
showRemoveFileModal: true
}

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    const showRemoveFileModal = true;
+    const onRemoveFile = sinon.spy();
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:13]: do you need the spys here...doesnt look like there are tests around them so id just remove them.

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    );
+    expect(
+      $('va-modal', container).getAttribute('secondarybuttontext'),
+    ).to.equal('No, keep this');
+  });
+
+  it('shows the filename to be removed', () => {
+    const removeFileName = 'test.pdf';
+    const showRemoveFileModal = true;
+    const onRemoveFile = sinon.spy();
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:37]: do you need the spys here...doesnt look like there are tests around them so id just remove them.

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    );
+
+    expect(getByText(removeFileName)).to.exist;
+  });
+
+  it('calls removeFile when primary button is clicked', () => {
+    const removeFileName = 'test.pdf';
+    const showRemoveFileModal = true;
+    const onRemoveFile = sinon.spy();
+    const onCloseModal = sinon.spy();
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:56]: do you need the onCloseModal spy here...doesnt look like there are tests around it so id just remove it.

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    );
+
+    $('va-modal', container).__events.primaryButtonClick();
+    expect(onRemoveFile.calledOnce).to.be.true;
+  });
+
+  it('calls closeModal when secondary button is clicked', () => {
+    const removeFileName = 'test.pdf';
+    const showRemoveFileModal = true;
+    const onRemoveFile = sinon.spy();
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:74]: do you need the onRemoveFile spy here...doesnt look like there are tests around it so id just remove it.

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    );
+
+    $('va-modal', container).__events.secondaryButtonClick();
+    expect(onCloseModal.calledOnce).to.be.true;
+  });
+
+  it('calls closeModal when close button is clicked', () => {
+    const removeFileName = 'test.pdf';
+    const showRemoveFileModal = true;
+    const onRemoveFile = sinon.spy();
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:76]: do you need the onRemoveFile spy here...doesnt look like there are tests around it so id just remove it.

File: src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { $ } from '@department-of-veterans-affairs/platform-forms-system/ui';
+
+import RemoveFileModal from '../../../components/claim-files-tab/RemoveFileModal';
+
+describe('<RemoveFileModal>', () => {
+  const props = {
+    removeFileName: 'test.pdf',
+    showRemoveFileModal: true,
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/claim-files-tab/RemoveFileModal.unit.spec.jsx:12]: you can set closeModal and removeFile here as well so you dont have to set it in the tests. 
EX:
closeModal: () => {},
removeFile: () => {},

**Review by pmclaren19 — APPROVED** [review]
LGTM!

**Pull request merged**